src/config_default_credentials.rs

```diff
@@ -30,9 +34,8 @@ impl ConfigDefaultCredentials {
     }
 
     pub(crate) async fn with_client(client: &HttpClient) -> Result<Self, Error> {
-        debug!("try to load credentials from {}", USER_CREDENTIALS_PATH);
-        let mut home = home::home_dir().ok_or(Error::Str("home directory not found"))?;
-        home.push(USER_CREDENTIALS_PATH);
+        let home = get_user_credentials_path()?;
+        debug!("try to load credentials from {:?}", home);
```

Comment:
Maybe it would be better to split this into two `debug!()` calls? I'd like to have some logging that announces trying the `ConfigDefaultCredentials` method that happens anyway, even if we fail to get the home directory (on Unix) or the `APPDATA` environment variable.

Also, bit of a pre-existing issue, we should probably use structured logging instead of formatting the directory into the log message (so use `path = home` before the message in the `debug!()` calls).

So maybe something like this?

```rust
debug!("try to load credentials from configuration");
let mut config_path = config_dir()?;
config_path.push(USER_CREDENTIALS_PATH);
debug!(config = config_path, "reading configuration file");
```